Add & use Token::text_view() (which returns a string_view unlike text())

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #151850
* #151849
* #151810
* #151807
* #151806
* #151805
* __->__ #151804
* #151803
* #151802
* #151801

Sadly, I can't just fix text() because that might cause lifetime issues in somebody's code.

Differential Revision: [D73376715](https://our.internmc.facebook.com/intern/diff/D73376715/)

cc @EikanWang @jgong5 @wenzhe-nrv @sanchitintel

swolchok committed:
Add & use Token::text_view() (which returns a string_view unlike text())

Sadly, I can't just fix text() because that might cause lifetime issues in somebody's code.

Differential Revision: [D73376715](https://our.internmc.facebook.com/intern/diff/D73376715/)

[ghstack-poisoned]

swolchok committed:
Update on "Add & use Token::text_view() (which returns a string_view unlike text())"


Sadly, I can't just fix text() because that might cause lifetime issues in somebody's code.

Differential Revision: [D73376715](https://our.internmc.facebook.com/intern/diff/D73376715/)

cc EikanWang jgong5 wenzhe-nrv sanchitintel

[ghstack-poisoned]

swolchok committed:
Update on "Add & use Token::text_view() (which returns a string_view unlike text())"


Sadly, I can't just fix text() because that might cause lifetime issues in somebody's code.

Differential Revision: [D73376715](https://our.internmc.facebook.com/intern/diff/D73376715/)

cc EikanWang jgong5 wenzhe-nrv sanchitintel

[ghstack-poisoned]

swolchok committed:
Update on "Add & use Token::text_view() (which returns a string_view unlike text())"


Sadly, I can't just fix text() because that might cause lifetime issues in somebody's code.

Differential Revision: [D73376715](https://our.internmc.facebook.com/intern/diff/D73376715/)

cc EikanWang jgong5 wenzhe-nrv sanchitintel

[ghstack-poisoned]

cyyever commented on torch/csrc/jit/frontend/lexer.h:
```diff
@@ -400,6 +400,11 @@ struct Token {
     return std::string(range.token_text());
   }
+
+  std::string_view text_view() {
```
Add `const` to the method?

cyyever commented on torch/csrc/jit/frontend/lexer.h:
```diff
@@ -400,6 +400,11 @@ struct Token {
   std::string text() {
```
Add const?

Skylion007 reviewed (approved):
Agree with the const comments

swolchok committed:
rebase and minor fixes as requested on "Add & use Token::text_view() (which returns a string_view unlike text())"


Sadly, I can't just fix text() because that might cause lifetime issues in somebody's code.

Differential Revision: [D73376715](https://our.internmc.facebook.com/intern/diff/D73376715/)

cc EikanWang jgong5 wenzhe-nrv sanchitintel

[ghstack-poisoned]

swolchok committed:
Update on "Add & use Token::text_view() (which returns a string_view unlike text())"


Sadly, I can't just fix text() because that might cause lifetime issues in somebody's code.

Differential Revision: [D73376715](https://our.internmc.facebook.com/intern/diff/D73376715/)

cc EikanWang jgong5 wenzhe-nrv sanchitintel

[ghstack-poisoned]
